.circleci/cimodel/data/windows_build_definitions.py

```diff
@@ -0,0 +1,137 @@
+        self.run_on_prs = vscode_spec.year != 2019
+
+    def gen_tree(self):
+
+        base_phase = "build" if self.test_index is None else "test"
+        numbered_phase = base_phase if self.test_index is None else base_phase + str(self.test_index)
+
+        key_name = "_".join(["pytorch", "windows", base_phase])
+
+        cpu_forcing_name_parts = ["on", "cpu"] if self.force_on_cpu else []
+
+        target_arch = self.cuda_version if self.cuda_version else "cpu"
+
+        base_name_parts = [
+            "pytorch",
+            "windows",
+            self.vscode_spec.render(),
+            "py36",
+            target_arch,
+        ]
+
+        prerequisite_jobs = []
+        if base_phase == "test":
+            prerequisite_jobs.append("_".join(base_name_parts + ["build"]))
+
+        arch_env_elements = ["cuda10", "cudnn7"] if self.cuda_version else ["cpu"]
+
+        build_environment_string = "-".join([
+            "pytorch",
+            "win",
+        ] + self.vscode_spec.get_elements() + arch_env_elements + [
+            "py3",
+        ])
+
+        is_running_on_cuda = bool(self.cuda_version) and not self.force_on_cpu
+
+        vc_product = "Community" if self.vscode_spec.year == 2019 else "BuildTools"
+
+        props_dict = {
+            "build_environment": build_environment_string,
+            "python_version": miniutils.quote("3.6"),
+            "vc_version": miniutils.quote(self.vscode_spec.dotted_version()),
+            "vc_year": miniutils.quote(str(self.vscode_spec.year)),
+            "vc_product": vc_product,
+            "use_cuda": miniutils.quote(str(int(is_running_on_cuda))),
+            "requires": ["setup"] + prerequisite_jobs,
+        }
+
+        if self.run_on_prs:
+            props_dict["filters"] = {
+                "branches": {
+                    "only": NON_PR_BRANCH_LIST,
+                },
+            }
+
+        name_parts = base_name_parts + cpu_forcing_name_parts + [
+            numbered_phase,
+        ]
+
+        if base_phase == "test":
+            test_name = "-".join(["pytorch", "windows", numbered_phase])
+            props_dict["test_name"] = test_name
+
+            if is_running_on_cuda:
+                props_dict["executor"] = "windows-with-nvidia-gpu"
+
+        props_dict["cuda_version"] = miniutils.quote(str(10)) if self.cuda_version else "cpu"
+        props_dict["name"] = "_".join(name_parts)
+
+        return [{key_name: props_dict}]
+
+
+class VcSpec:
+    def __init__(self, year, version_elements=[]):
+        self.year = year
+        self.version_elements = version_elements
+
+    def get_elements(self):
+        return [self.prefixed_year()] + self.version_elements
+
+    def dotted_version(self):
+        return ".".join(self.version_elements)
+
+    def prefixed_year(self):
+        return "vs" + str(self.year)
+
+    def render(self):
+        return "_".join(filter(None, [self.prefixed_year(), self.dotted_version()]))
+
+
+WINDOWS_WORKFLOW_DATA = [
```

Comment:
Are you sure we don't have a single cuda10.2 Windows config?

Reply:
100%.  This is another good reason to use code-generation for the yaml config instead of hand-writing it; it facilitates a centralized location for key data where we can review for supposed omissions.

In developing this pull request, I have verified that the generated code is identical to a normalized (key-sorted) version of the hand-written config.